lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -2462,6 +2462,98 @@ public void detach_message_to_released_channel_is_dropped() throws AblyException
         }
     }
 
+    /*
+     * Spec: RTN11d
+     * Checks that all channels become if the state is CLOSED transitions all the channels to
+     * INITIALIZED and unsets:
+     *  - RealtimeChannel.errorReason
+     *  - Connection.errorReason
+     *  - msgSerial
+     */
+    @Test
+    public void connect_on_closed_client_should_reinitialize_channels() throws AblyException {
+        ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+        try (AblyRealtime ably = new AblyRealtime(opts)) {
+
+            /* wait until connected */
+            (new ConnectionWaiter(ably.connection)).waitFor(ConnectionState.connected);
+            assertEquals("Verify connected state reached", ably.connection.state, ConnectionState.connected);
+
+            /* create a channel and attach */
+            final Channel channel = ably.channels.get("channel");
+            channel.attach();
+            new ChannelWaiter(channel).waitFor(ChannelState.attached);
+            assertEquals("Verify attached state reached", channel.state, ChannelState.attached);
+
+            /* push a message to increase msgSerial */
+            channel.publish("test", "test");
+            assertEquals(1, ably.connection.connectionManager.msgSerial);
+
+            ably.close();
+            new ChannelWaiter(channel).waitFor(ChannelState.detached);
+
```

Comment:
check for connection closed state and msgserial != 0

Reply:
not sure these are important assertions for this particular test, but added